File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_python/p03630/p03630_num7_parsed.py
# Let's find the maximum rectangle area
# Time Complexity: O(H * W)

def maximum_rectangle_area(h, w, grid):
    max_area = 0
    # Initialize row and column arrays with all values set to zero
    row = [0] * w
    col = [0] * h

    # Traverse each row and column to update the row and column arrays
    for i in range(h):
        for j in range(w):
            if grid[i][j] == '#':
                row[j] += 1
                col[i] += 1

    # Calculate the maximum area using the row and column arrays
    for i in range(h):
        for j in range(w):
            if grid[i][j] == '#':
                area = (row[j] + col[i]) - 1
                max_area = max(max_area, area)

    # Return the maximum possible area
    return max_area

# Input
h, w = map(int, input().split())
grid = [input() for _ in range(h)]

# Function call and output
print(maximum_rectangle_area(h, w, grid))

File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_cpp/p03630/p03630_num7_parsed.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

int main() {
    int H, W;
    cin >> H >> W;

    vector<string> grid(H);
    for (int i = 0; i < H; i++) {
        cin >> grid[i];
    }

    vector<vector<int>> dph(H, vector<int>(W, 0));
    vector<vector<int>> dpw(H, vector<int>(W, 0));

    // Dynamic programming for the longest consecutive black squares in each row and each column
    for (int i = 0; i < H; i++) {
        int cnt = 0;
        for (int j = 0; j < W; j++) {
            if (grid[i][j] == '#') {
                cnt++;
            } else {
                cnt = 0;
            }
            dph[i][j] = cnt;
        }
    }
    for (int j = 0; j < W; j++) {
        int cnt = 0;
        for (int i = 0; i < H; i++) {
            if (grid[i][j] == '#') {
                cnt++;
            } else {
                cnt = 0;
            }
            dpw[i][j] = cnt;
        }
    }

    int ans = 0;
    // Calculate the maximum possible area of Snuke's rectangle
    for (int i = 0; i < H; i++) {
        for (int j = 0; j < W; j++) {
            int area = dph[i][j] + dpw[i][j] - 1;
            ans = max(ans, area);
        }
    }

    cout << ans << endl;

    return 0;
}
